concordium-consensus/src/Concordium/Queries.hs

```diff
@@ -1005,9 +1008,38 @@ getBlockFinalizationSummary = liftSkovQueryBHI getFinSummarySkovM (\_ -> return
 getNextUpdateSequenceNumbers :: forall finconf. BlockHashInput -> MVR finconf (BHIQueryResponse NextUpdateSequenceNumbers)
 getNextUpdateSequenceNumbers = liftSkovQueryStateBHI query
   where
+    -- Get the next sequence number or 1, if not supported.
+    mNextSequenceNumber :: UQ.OUpdateQueue pt cpv e -> U.UpdateSequenceNumber
+    mNextSequenceNumber NoParam = 1
+    mNextSequenceNumber (SomeParam q) = UQ._uqNextSequenceNumber q
     query bs = do
         updates <- BS.getUpdates bs
-        return $ updateQueuesNextSequenceNumbers $ UQ._pendingUpdates updates
+        let UQ.PendingUpdates{..} = UQ._pendingUpdates updates
+        return
+            NextUpdateSequenceNumbers
+                { _nusnRootKeys = UQ._uqNextSequenceNumber _pRootKeysUpdateQueue,
+                  _nusnLevel1Keys = UQ._uqNextSequenceNumber _pLevel1KeysUpdateQueue,
+                  _nusnLevel2Keys = UQ._uqNextSequenceNumber _pLevel2KeysUpdateQueue,
+                  _nusnProtocol = UQ._uqNextSequenceNumber _pProtocolQueue,
+                  _nusnElectionDifficulty = mNextSequenceNumber _pElectionDifficultyQueue,
+                  _nusnEuroPerEnergy = UQ._uqNextSequenceNumber _pEuroPerEnergyQueue,
+                  _nusnMicroCCDPerEuro = UQ._uqNextSequenceNumber _pMicroGTUPerEuroQueue,
+                  _nusnFoundationAccount = UQ._uqNextSequenceNumber _pFoundationAccountQueue,
+                  _nusnMintDistribution = UQ._uqNextSequenceNumber _pMintDistributionQueue,
+                  _nusnTransactionFeeDistribution = UQ._uqNextSequenceNumber _pTransactionFeeDistributionQueue,
+                  _nusnGASRewards = UQ._uqNextSequenceNumber _pGASRewardsQueue,
+                  _nusnPoolParameters = UQ._uqNextSequenceNumber _pPoolParametersQueue,
+                  _nusnAddAnonymityRevoker = UQ._uqNextSequenceNumber _pAddAnonymityRevokerQueue,
+                  _nusnAddIdentityProvider = UQ._uqNextSequenceNumber _pAddIdentityProviderQueue,
+                  _nusnCooldownParameters = mNextSequenceNumber _pCooldownParametersQueue,
+                  _nusnTimeParameters = mNextSequenceNumber _pTimeParametersQueue,
+                  _nusnTimeoutParameters = mNextSequenceNumber _pTimeoutParametersQueue,
+                  _nusnMinBlockTime = mNextSequenceNumber _pMinBlockTimeQueue,
+                  _nusnBlockEnergyLimit = mNextSequenceNumber _pBlockEnergyLimitQueue,
+                  _nusnFinalizationCommitteeParameters = mNextSequenceNumber _pFinalizationCommitteeParametersQueue,
+                  _nusnValidatorScoreParameters = mNextSequenceNumber _pValidatorScoreParametersQueue,
+                  _nusnProtocolLevelTokensParameters = maybeWhenSupported minUpdateSequenceNumber id (UQ._pltUpdateSequenceNumber updates)
```

Comment:
Isn't this just the same as `mNextSequenceNumber (UQ._pltUpdateSequenceNumber updates)`?

Reply:
Almost, `mNextSequenceNumber` takes a queue, so it cannot be reused here